cassovary-core/src/main/scala/com/twitter/cassovary/graph/Traverser.scala

```diff
@@ -396,7 +396,7 @@ trait QueueBasedTraverser extends Traverser {
 /**
  * A traverser that keeps track of first depth of visiting a given node.
  */
-trait DepthTracker extends QueueBasedTraverser {
+trait DepthTracker[NodeType <: Node] extends QueueBasedTraverser[NodeType] {
```

Comment:
but DepthTracker only needs QueueBasedTraverser[Node] ? If so, okay to not
have type parametrization in DepthTracker.

On Wed, Feb 4, 2015 at 8:39 AM, Szymon <notifications@github.com> wrote:

> In
> cassovary-core/src/main/scala/com/twitter/cassovary/graph/Traverser.scala
> <https://github.com/twitter/cassovary/pull/148#discussion_r24097040>:
>
> > @@ -396,7 +396,7 @@ trait QueueBasedTraverser extends Traverser {
> >  /**
> >   * A traverser that keeps track of first depth of visiting a given node.
> >   */
> > -trait DepthTracker extends QueueBasedTraverser {
> > +trait DepthTracker[NodeType <: Node] extends QueueBasedTraverser[NodeType] {
>
> QueueBasedTraverser should definitely be of NodeType.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twitter/cassovary/pull/148/files#r24097040>.
>